spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java

```diff
@@ -96,9 +94,39 @@ public final ColumnarBatch read(ColumnarBatch reuse, int numRowsToRead) {
       Integer numRows = rowIdMapping.second();
       batch.setNumRows(numRows);
     }
+
+    if (deletes != null && deletes.hasEqDeletes()) {
+      applyEqDelete(batch, rowIdMapping == null ? null : rowIdMapping.first(), eqDeleteRowIdMapping);
```

Comment:
I would probably try to extract out the logic for choosing which row mapping to use from "applyEqDelete" and do it here instead so it's simpler to read.